beater/beater.go

```diff
@@ -192,11 +194,27 @@ func initTracer(info beat.Info, config *Config, logger *logp.Logger) (*elasticap
 	if err != nil {
 		return nil, nil, err
 	}
-	if config.SelfInstrumentation.isEnabled() {
-		if config.SelfInstrumentation.Environment != nil {
-			tracer.Service.Environment = *config.SelfInstrumentation.Environment
+	// tracing disabled, setup complete
+	if !config.SelfInstrumentation.isEnabled() {
+		return tracer, nil, nil
+	}
+
+	if config.SelfInstrumentation.Environment != nil {
+		tracer.Service.Environment = *config.SelfInstrumentation.Environment
+	}
+	tracer.SetLogger(logp.NewLogger("tracing"))
+
+	// tracing destined for external host
+	// only first host used until https://github.com/elastic/apm-agent-go/issues/200
+	if config.SelfInstrumentation.Hosts != nil {
+		t, err := transport.NewHTTPTransport(config.SelfInstrumentation.Hosts[0], config.SelfInstrumentation.SecretToken)
+		if err != nil {
+			return nil, nil, err
```

Comment:
In https://github.com/elastic/apm-server/pull/1389/files#diff-110fdf0113b0dcfcdca7c9352c1b4c4fR228 `tracer.Close()` is called in an error case. You don't call it here, but return `nil` instead of the tracer, so I think it would never get closed in this case.

Reply:
Great call, fixed